mmselfsup/models/backbones/vision_transformer.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import math
+from functools import reduce
+from operator import mul
+
+import torch.nn as nn
+from mmcls.models.backbones import VisionTransformer as _VisionTransformer
+from mmcls.models.utils import PatchEmbed, to_2tuple
+from torch.nn.modules.batchnorm import _BatchNorm
+
+from mmselfsup.models.utils import build_2d_sincos_position_embedding
+from ..builder import BACKBONES
+
+
+@BACKBONES.register_module()
+class VisionTransformer(_VisionTransformer):
+    """Vision Transformer.
+
+    A pytorch implement of: `An Images is Worth 16x16 Words: Transformers for
+    Image Recognition at Scale <https://arxiv.org/abs/2010.11929>`_.
+
+    Part of the code is modified from:
+    `<https://github.com/facebookresearch/moco-v3/blob/main/vits.py>`_.
+
+    Args:
+        stop_grad_conv1 (bool, optional): whether to stop the gradient of
+            convolution layer in `PatchEmbed`. Defaults to False.
+        frozen_stages (int): Stages to be frozen (stop grad and set eval mode).
+            -1 means not freezing any parameters. Defaults to -1.
+        norm_eval (bool): Whether to set norm layers to eval mode, namely,
+            freeze running stats (mean and var). Note: Effect on Batch Norm
+            and its variants only. Defaults to False.
+        init_cfg (dict or list[dict], optional): Initialization config dict.
+            Defaults to None.
+    """
+
+    arch_zoo = {
+        **dict.fromkeys(
+            ['s', 'small'], {
+                'embed_dims': 384,
+                'num_layers': 12,
+                'num_heads': 12,
+                'feedforward_channels': 1536,
+            }),
+        **dict.fromkeys(
```

Comment:
`Visiontransformer` in `mmcls` already contains this attribute, considering removing it for simplicity.

Reply:
the `small` arch is diffenrent from mmcls